File: java-home-inventory-core/src/br/com/sailboat/homeinventory/core/interactor/product/GetProducts.java
package br.com.sailboat.homeinventory.core.interactor.product;

import java.util.ArrayList;
import java.util.List;

import br.com.sailboat.homeinventory.core.Filter;
import br.com.sailboat.homeinventory.core.entity.Product;
import br.com.sailboat.homeinventory.core.interactor.UseCase;
import br.com.sailboat.homeinventory.core.repository.ProductRepository;

public class GetProducts implements UseCase<List<Product>> {

    private ProductRepository productRepository;
    private Filter filter;

    public GetProducts(ProductRepository productRepository, Filter filter) {
        this.productRepository = productRepository;
        this.filter = filter;
    }

    @Override
    public List<Product> execute() {
        List<Product> products = productRepository.getAll(filter);

        if (products == null) {
            products = new ArrayList<>();
        }

        return products;
    }

}

File: android-home-inventory-app/app/src/main/kotlin/br/com/sailboat/homeinventory/domain/usecase/product/GetProducts.kt
package br.com.sailboat.homeinventory.domain.usecase.product

import br.com.sailboat.homeinventory.domain.None
import br.com.sailboat.homeinventory.domain.entity.Product
import br.com.sailboat.homeinventory.domain.repository.ProductRepository
import br.com.sailboat.homeinventory.domain.usecase.UseCase
import javax.inject.Inject

class GetProducts @Inject constructor(private val productRepository: ProductRepository) : UseCase<None, List<Product>>() {

    override fun execute(params: None) = productRepository.getProductList()

}